icu4c/source/i18n/unicode/numfmt.h

```diff
@@ -1077,7 +1078,7 @@ class U_I18N_API NumberFormat : public Format {
      * other classes have different class IDs.
      * @stable ICU 2.0
      */
-    virtual UClassID getDynamicClassID(void) const = 0;
+    //virtual UClassID getDynamicClassID(void) const = 0;
```

Comment:
This should not be removed. Once removed, the build/compile time error reporting of the subclass of NumberFormat w/o implement getDynamicClassID would be removed and may have a subclass w/o implement such.

Reply:
virtual UClassID getDynamicClassID(void) const override = 0;